### DISCUSSION TITLE
RavenDB F# status

### DISCUSSION THREAD

**Discussion by Peluko** [ravendb/ravendb#14701] [Q&A] [answered]
Hello. RavenDB has just come into my radar and for what I've read it could fit in a project I'm planning now.

The code base for the project is mainly in F#, but I can't find any specific F# docs or client. The only docs I can find are these largely outdated: https://github.com/ravendb/docs/blob/master/Documentation/1.0/Raven.Documentation.Pages/client-api/fsharp.markdown

I suppose that I can use the C# client directly, but my question is:

What's the status for RavenDB on F#? It just works or may I encounter problems? If it works, does it work on a F#-way or it requires to go down and do it the C#-way using F#?

Regards.

**Comment by ayende** [comment, marked as answer]
In general, everything should Just Work.

There are certain cases that need attention, is you are using some of the F# features. Discriminated Unions, for example, may require that you'll customize the serialization if you want them to be idiomatic JSON rather than whatever JSON.Net does.

Nothing special about RavenDB, mind, just normal F# -> JSON stuff.

**Comment by DejanMilicic** [comment]
You will have to use `[<CLIMutable>]` on your types in order to properly load the from RavenDB.

The only problem I ran into so far was translating F# Quotations into C# Expressions, but this is language-related issue, not RavenDB one
